src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,239 @@
+            },
+            1...16 => try self.script.append(@intCast(Opcode.OP_1.toBytes() - 1 + num)),
+            else => {
+                _ = try self.addData(try ScriptNum.new(num).toBytes(self.allocator));
+            },
+        }
+        return self;
+    }
+
+    pub fn addData(self: *ScriptBuilder, data: []const u8) !*ScriptBuilder {
+        if (data.len == 0 or data.len == 1 and data[0] == 0) {
+            _ = try self.addOpcode(Opcode.OP_0);
+        } else if (data.len == 1 and data[0] <= 16) {
+            const op = Opcode.OP_1.toBytes() - 1 + data[0];
+            try self.script.append(op);
+        } else if (data.len == 1 and data[0] == 0x81) {
+            _ = try self.addOpcode(Opcode.OP_1NEGATE);
+        } else if (data.len < Opcode.OP_PUSHDATA1.toBytes()) {
+            _ = try self.script.append(@intCast(data.len));
+            for (data) |byte| {
+                try self.script.append(byte);
+            }
+        } else if (data.len >= Opcode.OP_PUSHDATA1.toBytes()) {
+            if (data.len > 75 and data.len <= 255) {
+                _ = try self.script.append(76);
+                const size = &[1]u8{@intCast(data.len)};
+                const n = std.mem.readInt(u8, size, .little);
+                _ = try self.script.append(n);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            } else if (data.len >= 256 and data.len <= 65535) {
+                _ = try self.script.append(77);
+                const length: u16 = @intCast(data.len);
+                const n: [2]u8 = [2]u8{
+                    @intCast(length & 0xFF), // lower byte (least significant byte)
+                    @intCast((length >> 8) & 0xFF), // upper byte (most significant byte)
+                };
+                _ = try self.script.append(n[0]);
+                _ = try self.script.append(n[1]);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            } else if (data.len > 65535 and data.len <= 4294967295) {
+                _ = try self.script.append(78);
+                const length: u32 = @intCast(data.len);
+                const n: [4]u8 = [4]u8{
+                    @intCast(length & 0xFF), // lower byte (least significant byte)
+                    @intCast(length >> 8 & 0xFF), // next byte
+                    @intCast(length >> 16 & 0xFF), // next byte
+                    @intCast((length >> 24) & 0xFF), // upper byte (most significant byte)
+                };
+                _ = try self.script.append(n[0]);
+                _ = try self.script.append(n[1]);
+                _ = try self.script.append(n[2]);
+                _ = try self.script.append(n[3]);
+
+                for (data) |byte| {
+                    try self.script.append(byte);
+                }
+            }
+        } else {
+            _ = try self.script.append(Opcode.OP_PUSHDATA1.toBytes());
+        }
```

Comment:
wtf is that for?
There is not `else` case, you have covered all the possible values of an i32 by now